src/generic/FreeModule.jl

```diff
@@ -56,6 +56,13 @@ end
 
 basis(N::FreeModule) = gens(N)
 
+function Base.hash(a::FreeModuleElem, h::UInt)
+   b = 0x3a1fe8d97f5f372e%UInt
+   b = xor(b, xor(hash(a.v, h), h))
+   b = (b << 1) | (b >> (sizeof(Int)*8 - 1))
+   return b
+end
```

Comment:
What is the advantage of this over
```suggestion
Base.hash(a::FreeModuleElem, h::UInt) = hash(a.v, u)
```

Reply:
But how can that happen unless you do something stupid and use a `Dict{Any,...}` with keys of *different* types?

IMHO this solves a non-problem (or worse, it hides a symptom caused by a deeper issue)